Core/GDCore/Project/Layer.cpp

```diff
@@ -13,7 +14,11 @@ namespace gd {
 Camera Layer::badCamera;
 Effect Layer::badEffect;
 
-Layer::Layer() : isVisible(true), isLightingLayer(false), followBaseLayerCamera(false) {}
+Layer::Layer()
+    : isVisible(true),
+      isLightingLayer(false),
+      followBaseLayerCamera(false),
+      effectsContainer(std::make_unique<EffectsContainer>()) {}

```

Comment:
Is there a need to initialize it? Or is it already initialized like `std::vector`? Also, should it be a unique_ptr or a shared_ptr? As I could understand, EffectsContainer and Layer have a composition relationship, so I'm not sure if an EffectContainer object will be shared or not 🤔.

Reply:
As said in the other comment, remove the usage of a unique ptr. Because this is a part of the class (a layer always has a container of effects (even if this container is empty)), we don't need a pointer, we're happy to have it as a traditional member.
This being said, the container itself is using shared_ptr, so that when we re-organise the order of effects, the pointers are staying valid :)